submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestReleaser.kt

```diff
@@ -111,6 +118,54 @@ class SubmissionRequestReleaser(
         return releasedFile
     }
 
+    private suspend fun suppressRequest(
+        accNo: String,
+        request: SubmissionRequest,
+    ) {
+        val sub = request.submission
+        logger.info { "$accNo ${sub.owner} Started suppressing submission files over ${sub.storageMode}" }
+        suppressSubmissionFiles(sub)
+        logger.info { "$accNo ${sub.owner} Finished suppressing submission files over ${sub.storageMode}" }
+    }
+
+    private suspend fun suppressSubmissionFiles(sub: ExtSubmission) {
+        suspend fun suppressFile(reqFile: SubmissionRequestFile) {
+            when (val file = reqFile.file) {
+                is NfsFile -> {
+                    val released = reqFile.copy(file = suppress(sub, reqFile.index, file), status = SUPPRESSED)
+                    rqtService.updateRqtFile(released)
+                }
+
+                is FireFile -> {
+                    if (file.published) {
+                        val suppressed = reqFile.copy(file = suppress(sub, reqFile.index, file), status = SUPPRESSED)
+                        rqtService.updateRqtFile(suppressed)
+                    } else {
+                        rqtService.updateRqtFile(reqFile.copy(status = SUPPRESSED))
+                    }
+                }
+            }
+        }
+
+        supervisorScope {
+            filesRequestService
+                .getSubmissionRequestFiles(sub.accNo, sub.version, COPIED)
```

Comment:
this is really confusing for me because we need to unrelease (AKA supress) old submission files but we are iterating over new submission files. 

it looks to me that we need a new stage only executed if there is an old version exists is released and new submission is not released. I guess we can create this new stage after my changes which basically can index those files identified as RE_USED.